forem/forem
# UsersSettings RFC: Code to Sync User Settings updates to new tables

## What type of PR is this? (check all applicable)


## Description
- This PR accomplishes Action Step 3 of the [UsersSettings RFC](https://github.com/forem/rfcs/pull/32#issuecomment-820442032).
- For a newly created user, the code creates one Users::Setting record and one Users::NotificationSetting for that user, and then populates the relevant fields in these records with the User table values for that user.
- When a user is updated, the code populates that user's existing `Users::Setting` or `Users::NotificationSetting` record.


## Related Tickets & Documents
Data Update Script PR: https://github.com/forem/forem/pull/13390

https://github.com/forem/rfcs/pull/32

## QA Instructions, Screenshots, Recordings

1. On the front-end, create a new user (do not use the console, so all relevant callbacks can fire).
2. Thereafter, run the command `Users::Setting.find_by(user_id: id-of-created-user)`. You should get a Users::Setting record back. Check that the attributes in this record match those for the user (eg `User.find(id-of-created-user).attribute_name`).
4. Go through the same flow, but this time update a relevant field for an existing user. That attribute's value should also be updated in the `Users::Setting` record for that user.
5. Repeat the above steps for the user's `Users::NotificationSetting` record.

### UI accessibility concerns?
None. This is all backend work.



## Added tests?

- [ ] No, not yet



## [Forem core team only] How will this change be communicated?

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_



## [optional] What gif best describes this PR or how it makes you feel?

![Pooh Bear Stretching](https://media.giphy.com/media/JPbZwjMcxJYic/giphy.gif)
